Source/player.cpp

```diff
@@ -367,27 +367,75 @@ bool PlayerStruct::IsWalking() const
 	}
 }
 
-void SetPlayerGPtrs(byte *pData, byte **pAnim)
+void PlayerStruct::Reset()
 {
-	int i;
+	// Create empty default initialized PlayerStruct on heap to avoid excessive stack usage
+	auto emptyPlayer = std::make_unique<PlayerStruct>();
+	*this = std::move(*emptyPlayer);

```

Comment:
No need for a variable here?

```suggestion
	*this = std::make_unique<PlayerStruct>();
```

Reply:
`error C2679: binary '=': no operator found which takes a right-hand operand of type 'std::unique_ptr<devilution::PlayerStruct,std::default_delete<devilution::PlayerStruct>>' (or there is no acceptable conversion)`

`*this = *std::make_unique<PlayerStruct>();`:
I get `error C2280: 'devilution::PlayerStruct &devilution::PlayerStruct::operator =(const devilution::PlayerStruct &)': attempting to reference a deleted function`

`*this = std::move(*std::make_unique<PlayerStruct>());` compiles
But this looks confusing :hushed:.

Should we define a normal copy constructor?